fineract-loan/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/transactionprocessor/AbstractLoanRepaymentScheduleTransactionProcessor.java

```diff
@@ -395,6 +397,16 @@ protected void reprocessInstallments(List<LoanRepaymentScheduleInstallment> inst
             if (loanRepaymentScheduleInstallment.getTotalOutstanding(currency).isGreaterThanZero()) {
                 loanRepaymentScheduleInstallment.updateObligationMet(false);
                 loanRepaymentScheduleInstallment.updateObligationMetOnDate(null);
+            } else {
+                Optional<LoanTransaction> optWaiverTx = transactions.stream().filter(lt -> {
+                    LocalDate fromDate = loanRepaymentScheduleInstallment.getFromDate();
+                    return lt.getSubmittedOnDate().isAfter(fromDate);
```

Comment:
i think you need to use the transaction date here.

Reply:
done.